### ISSUE TITLE
Modification d'une observation et statut de validation

### ISSUE THREAD

**Issue by camillemonchicourt** [PnX-SI/GeoNature#514]
Déplacement de la discussion sur le sujet depuis : https://github.com/PnX-SI/GeoNature/issues/513

@DonovanMaillard : 
Il se passe quoi quand on update une donnée? Elle reste valide?
Je saisie un truc qui pourrait etre cohérent, on valide, je modifie en mettant une grosse connerie, est-ce que le statut de validation est conservé?
La difficulté, come from naturaliste ;) , c'est que ça devrait dépendre des champs :D
j'ajoute un sexe un dénombrement, ca change pas la validation. Je change l'espèce, le déterminateur, la méthode de détermination... ba ça dépend :D Tu le sens le truc chiant qui arrive à petit pas?

@camillemonchicourt : 
Bof...

@sig-pnrnm : 
Pour ma part, je partirais du principe que, dès qu'une occurrence est modifiée, quelle que soit la modification (sinon, on n'en fini plus), elle repasse en statut "en attente de validation" (j'ai oublié le terme exact).
Mais comme, en principe, on garde l'historique des différents statuts de validation, cette info sera disponible quand le module de validation sera opérationnel.

@gildeluermoz : 
Autant je comprends la logique de ta proposition Donovan, autant je dirais : Oulala ! J'aimerais pas être validateur. Tu valides qq chose mais c'est pas définitif, faut y revenir... Encore et encore. Tout ça parce que machin chouette qui à les droit a corrigé une faute d'orthographe dans les commentaires ou a modifié de qq mètre l'altitude, parce que quand même faut être précis ! Et là le validateur va dire : "Mais je l'ai déjà validée cette obs !"
Et moi je vous dirai : "mais qu'elle usine à gaz cette notion de validation !".

@sig-pnrnm : 
Personnellement, je pense que c'est moins "usine à gaz" si on part du principe que toute modification implique re-validation. Si on doit identifier les modification qui implique modification ou non, là on tombe vraiment dans l'usine à gaz !

Sinon, si je comprends que ce sujet soit complexe à modéliser d'un point de vue informatique. Mais je reviens d'une réunion dans le cadre de notre plateforme SINP régionale, avec tous les acteurs naturalistes Normands autour de la table, et cet impératif de validation scientifique des données est revenu à plusieurs reprises.
Notamment, en citant le site de l'INPN qui affiche maintenant, avec la remontée croissante de données dans le cadre SINP, des données clairement fausses...

**Comment by DonovanMaillard** [comment]
D'accord avec toi Gil, de toutes facons j'aimerais pas être validateur tout cours ;) 

Mais je pense que le principe même de validation, ne serait-ce que niveau SINP, peut être pas mal critiqué si un observateur peut modifier son observation après qu'elle ait été, ou non, invalidée, sans que ça en change le statut.

Deux exemples, 
- une guguerre entre validateur et un naturaliste qui pense avoir raison, et qui remettra son truc invalidé après validation... (mine de rien, pas mal entendu de désaccords du genre par ceux dont les données font l'objet de validation). 
- une donnée invalidée, corrigée par l'observateur qui a vu sa faute avec le validateur, mais qui automatiquement reste invalidée... 
Dans les deux cas, le fonctionnement risque de faire grincer les dents. 

Après, niveau développement oui, ca devient machine à gaz... mais niveau efficacité, on revient sur la validation sauf si les champs sans incidence sont corrigés comme le commentaire justement, c'est pas forcément absurde (à mon sens :) )

**Comment by DonovanMaillard** [comment]
ps : pourquoi ça se discute ici et pas dans gn_module_validation? :)

**Comment by gildeluermoz** [comment]
Oui, je suis d'accord, à déplacer.

**Comment by camillemonchicourt** [comment]
On pourrait en discuter dans le dépôt du module Validation, mais c'est un sujet assez global à GeoNature la validation des observations et lié au cœur donc ça va bien ici aussi.

**Comment by sig-pnrnm** [comment]
Oui, pas de souci à en discuter ici. Mais il faudrait référencer cette issue sur le dépot du module.
Je vais tenter de le faire, mais si vous voulez le faire plus proprement ailleurs, n'hésitez pas

**Comment by sig-pnrnm** [comment]
> quelle que soit la modification [...], elle repasse en statut "en attente de validation" (j'ai oublié le terme exact)

Maintenant que j'ai l'info, j'apporte la précision : il s'agit du statut `Non évalué` du standard, ou `En attente de validation` (`id_nomenclature = 466`) dans `t_nomenclature`.
Ca colle bien pour moi, pour une donnée à réexaminer.

A moins que l'on créé un statut spécifique : `donnée validée modifiée : à vérifier`

**Comment by camillemonchicourt** [comment]
Faut voir. Ajouter une nomenclature hors SINP est déconseillé.
De toute chacun peut définir une valeur de nomenclature par défaut donc la BDD donc on pourrait utiliser celle-ci.

**Comment by DonovanMaillard** [comment]
Sujet abordé de nombreuses fois, sans réponse idéale. Aucune modification planifiée et pas de meilleure solution identifiée à ce stade, on indique simplement qu'une donnée a été modifiée depuis sa dernière validation.
